component/org.wso2.carbon.identity.casque.authenticator/src/main/java/org/wso2/carbon/identity/casque/authenticator/authenticator/CasqueAuthenticator.java

```diff
@@ -0,0 +1,264 @@
+
+        if (radiusState == null) { // Initial request, get a challenge
+            AuthenticatorFlowStatus status = null;
+            try {
+                status = start(request, response, context);
+            } catch (AuthenticationFailedException e) {
+
+                throw new AuthenticationFailedException("Error Accord while sending request  ", e);
+            }
+            context.setCurrentAuthenticator(getName());
+            return status;
+        }
+        // radiusState is not null so handle the response to the challenge
+        context.setProperty(CasqueAuthenticatorConstants.RADIUS_STATE, null);
+
+        String action = request.getParameter(CasqueAuthenticatorConstants.BTN_ACTION);
+        if (action != null && "Login".equals(action)) { // action can be null, Login or Cancel
+
+            String userName = (String) context.getProperty(CasqueAuthenticatorConstants.USER_NAME);
+            String challengeResponse = request.getParameter(CasqueAuthenticatorConstants.RESPONSE);
+            try {
+                // Send the response to the CASQUE Server
+                RadiusResponse radiusResponse = Radius.sendRequest(userName, challengeResponse, radiusState);
+                int radiusResponseType = radiusResponse.getType();
+
+                if (radiusResponseType == RadiusResponse.ACCESS_CHALLENGE) { // Another challenge.
+                    context.setProperty(CasqueAuthenticatorConstants.RADIUS_STATE, radiusResponse.getState());
+                    String challenge = radiusResponse.getChallenge();
+                    String contextIdentifier = context.getContextIdentifier();
+                    authPages.challengePage(response, contextIdentifier, challenge);
+                    return AuthenticatorFlowStatus.INCOMPLETE;
+                }
+
+                clearProperties(context);
+
+                if (radiusResponseType == RadiusResponse.ACCESS_ACCEPT) { // Authentication Pass.
+                    context.setSubject(AuthenticatedUser.createLocalAuthenticatedUserFromSubjectIdentifier(userName));
+                    String tokenId = getCasqueTokenId(userName);
+                    log.debug("CASQUE Authentication PASS for " + userName + "with Token" + tokenId);
+                    request.setAttribute(FrameworkConstants.REQ_ATTR_HANDLED, true);
+                    return AuthenticatorFlowStatus.SUCCESS_COMPLETED;
+                }
+
+                if (radiusResponseType == RadiusResponse.ACCESS_REJECT) { // Authentication Failed.
+                    throw new InvalidCredentialsException("User authentication failed due to invalid credentials",
+                            User.getUserFromUserName(userName));
+                }
+
+                throw new InvalidCredentialsException("User authentication failed due to" +
+                        radiusResponse.getError(),
+                        User.getUserFromUserName(userName));
```

Comment:
Formate above three lines and  move this to the above line

Reply:
done